### ISSUE TITLE
Request: cross-platform websocket API

### ISSUE THREAD

**Issue by jkelleyrtp** [DioxusLabs/sdk#53]
I love using websockets, and frequently set them up for various projects, but there's no great websocket library that does both wasm and native.

I might be bad at sleuthing crates.io, but I can't find a ws library that has backoffs, reconnects, heartbeats, and works across platforms.

This one looks nice:
https://crates.io/crates/reconnecting-websocket

As part of our sdk we should provide a really nice websocket library that we can point people to when they inevitably need websockets for their crossplatform apps.

**Comment by marc2332** [comment]
I honestly don't see the point of doing this in the SDK. Don't get me wrong, the idea of having a simple library for WebSockets is a good idea, but it doesn't have to be connected to Dioxus at all. Maybe it could be done in a standalone repo under the DioxusLabs org. 

When it comes to actual integrations with Dioxus in the form of hooks or other ways, that could definitely be done in the SDK

**Comment by DogeDark** [comment]
I agree. I'd say similar for a lot of stuff that may go in SDK, like mobile features.

I've wanted to release geolocation as it's own crate but have avoided it as more crates is more overhead.
